packages/teleterm/src/mainProcess/protocolHandler.ts

```diff
@@ -0,0 +1,88 @@
+import type {
+  protocol as ElectronProtocol,
+  ProtocolRequest,
+  ProtocolResponse,
+} from 'electron';
+import * as path from 'path';
+import fs from 'fs';
+
+type CallbackType = (response: string | ProtocolResponse) => void;
+
+function _disabledHandler(
+  _request: ProtocolRequest,
+  callback: CallbackType
+): void {
+  callback({ error: -3 });
+}
+
+export function installWebHandler({
+  protocol,
+}: {
+  protocol: typeof ElectronProtocol;
+}): void {
+  protocol.interceptFileProtocol('about', _disabledHandler);
+  protocol.interceptFileProtocol('content', _disabledHandler);
+  protocol.interceptFileProtocol('chrome', _disabledHandler);
+  protocol.interceptFileProtocol('cid', _disabledHandler);
+  protocol.interceptFileProtocol('data', _disabledHandler);
+  protocol.interceptFileProtocol('filesystem', _disabledHandler);
+  protocol.interceptFileProtocol('ftp', _disabledHandler);
+  protocol.interceptFileProtocol('gopher', _disabledHandler);
+  protocol.interceptFileProtocol('javascript', _disabledHandler);
+  protocol.interceptFileProtocol('mailto', _disabledHandler);
+}
+
+function eliminateAllAfterCharacter(string, character) {
+  const index = string.indexOf(character);
+  if (index < 0) {
+    return string;
+  }
+
+  return string.slice(0, index);
+}
+
+function urlToPath(targetUrl, { isWin }) {
+  const decoded = decodeURIComponent(targetUrl);
+  const withoutScheme = decoded.slice(isWin ? 8 : 7);
```

Comment:
Could you add a comment explaining why `isWin` is needed here?

Reply:
Will add comment. For conversation, `isWin` is used to slice the files names slightly differently based on their shape.
Windows: `file:///C:/path/to/file`
unix: `file:///path/to/file`

slicing by 8 on windows is needed to get to that root `C:` drive (or whatever drive they need)

Reply:
FWIW, you could also do `new URL()` and then `url.pathname` so that you only need to get rid of a single character.

Reply:
I love this!